pkgs/development/compilers/gcc/8/default.nix

```diff
@@ -0,0 +1,519 @@
+{ stdenv, targetPackages, fetchurl, fetchpatch, noSysDirs
+, langC ? true, langCC ? true, langFortran ? false
+, langObjC ? targetPlatform.isDarwin
+, langObjCpp ? targetPlatform.isDarwin
+, langJava ? false
+, langGo ? false
+, profiledCompiler ? false
+, staticCompiler ? false
+, enableShared ? true
+, texinfo ? null
+, perl ? null # optional, for texi2pod (then pod2man); required for Java
+, gmp, mpfr, libmpc, gettext, which
+, libelf                      # optional, for link-time optimizations (LTO)
+, isl ? null # optional, for the Graphite optimization framework.
+, zlib ? null, boehmgc ? null
+, zip ? null, unzip ? null, pkgconfig ? null
+, gtk2 ? null, libart_lgpl ? null
+, libX11 ? null, libXt ? null, libSM ? null, libICE ? null, libXtst ? null
+, libXrender ? null, xproto ? null, renderproto ? null, xextproto ? null
+, libXrandr ? null, libXi ? null, inputproto ? null, randrproto ? null
+, x11Support ? langJava
+, enableMultilib ? false
+, enablePlugin ? hostPlatform == buildPlatform # Whether to support user-supplied plug-ins
+, name ? "gcc"
+, libcCross ? null
+, crossStageStatic ? false
+, libpthread ? null, libpthreadCross ? null  # required for GNU/Hurd
+, stripped ? true
+, gnused ? null
+, cloog # unused; just for compat with gcc4, as we override the parameter on some places
+, darwin ? null
+, buildPlatform, hostPlatform, targetPlatform
+, buildPackages
+}:
+
+assert langJava     -> zip != null && unzip != null
```

Comment:
I thought upstream had removed java support in gcc 6?

Reply:
Oh, yeah, right you are. Should the arguments stay in place and go unused (apart from maybe an `assert not langJava`), or should I remove them altogether (https://github.com/NixOS/nixpkgs/blob/ba52ae50488de85a9cf60a3a04f1c9ca7122ec74/pkgs/development/compilers/gcc/7/default.nix#L30 seems to suggest that the first option would be more conventional.)

Reply:
I think we should remove them completely.  AFAICT the only place where we override enableJava et al. is in the gcj6 attribute (which uses gcc6).